.github/workflows/validate-swagger.yml

```diff
@@ -0,0 +1,13 @@
+name: Validate API swagger documentation
```

Comment:
I would change this to "Documentation" and rename the file to "docs.yml". The name of a step should be as broad as possible, the information on what is currently doing is contained in the job.